test_common/harness/imageHelpers.cpp

```diff
@@ -2953,64 +2954,41 @@ int DetectFloatToHalfRoundingMode(
             return err;
         }
 
-        err = clBuildProgram(program, 1, &device, "", NULL, NULL);
-        if (err)
-        {
-            log_error("Error:  could not build program in "
-                      "DetectFloatToHalfRoundingMode  (%d)",
-                      err);
-            clReleaseMemObject(inBuf);
-            clReleaseMemObject(outImage);
-            return err;
-        }
-
-        cl_kernel k = clCreateKernel(program, "detect_round", &err);
-        if (NULL == k || err)
-        {
-            log_error("Error:  could not create kernel in "
-                      "DetectFloatToHalfRoundingMode  (%d)",
-                      err);
-            clReleaseMemObject(inBuf);
-            clReleaseMemObject(outImage);
-            return err;
-        }
-
-        err = clSetKernelArg(k, 0, sizeof(cl_mem), &inBuf);
+        err = clSetKernelArg(kernel, 0, sizeof(cl_mem), &inBuf);
         if (err)
         {
             log_error("Error: could not set argument 0 of kernel in "
                       "DetectFloatToHalfRoundingMode (%d)",
                       err);
             clReleaseMemObject(inBuf);
             clReleaseMemObject(outImage);
-            clReleaseKernel(k);
+            clReleaseKernel(kernel);
```

Comment:
`kernel` is now a wrapper. You should remove the calls to `clReleaseKernel` altogether. There are a few more occurrences below.